pkg/resources/nifi/pod.go

```diff
@@ -543,10 +546,32 @@ exec bin/nifi.sh run`, resolveIp, singleUser)}
 	}
 }
 
+func (r *Reconciler) injectAdditionalFields(containers []corev1.Container) []corev1.Container {
+	withEnvVars := r.injectAdditionalEnvVars(containers)
+	withSecurityContext := r.injectAdditionalSecurityContext(withEnvVars)
+	return withSecurityContext
+}
+
 func (r *Reconciler) injectAdditionalEnvVars(containers []corev1.Container) (injectedContainers []corev1.Container) {
 	for _, container := range containers {
 		container.Env = append(container.Env, r.NifiCluster.Spec.ReadOnlyConfig.AdditionalSharedEnvs...)
 		injectedContainers = append(injectedContainers, container)
 	}
 	return
 }
+
+func (r *Reconciler) injectAdditionalSecurityContext(containers []corev1.Container) (injectedContainers []corev1.Container) {
+	for _, container := range containers {
+		container.SecurityContext = &corev1.SecurityContext{
```

Comment:
Can we add an optional `corev1.SecurityContext` to the NifiCluster...NodeConfig` so it can be overridden from this default? We can default to dropping all capabilities as written here.